src/DynamoCore/Core/PulseMaker.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Threading;
+
+using Dynamo.Models;
+
+namespace Dynamo.Core
+{
+    public class PulseMaker
+    {
+        #region Class Data Members and Properties
+
+        private readonly Timer internalTimer;
+        private readonly object stateMutex;
+        private bool evaluationInProgress = false;
+        private bool evaluationRequestPending = false;
+
+        internal int TimerPeriod { get; private set; }
+
+        public event EventHandler<EventArgs> RunStarted;
```

Comment:
Will there ever be a situation where `PulseMaker` is NOT the sender and `EventArgs` is non-empty?  Why not just do `Action`?